torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3759,6 +3760,97 @@ def test_ddp_comm_hook_logging(self):
             # type if it didn't exist.
             self.assertEqual(ddp_logging_data.get("comm_hook", ""), "")
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "Only Nccl & Gloo backend support DistributedDataParallel",
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_hook_with_optimizer_parity(self):
+            rank = self.rank
+            torch.cuda.set_device(rank)
+            torch.manual_seed(rank)
+            torch.cuda.manual_seed(rank)
+            models_to_test = [
+                (LargeNet(), torch.randn(1, 1000).cuda()),
+                (torchvision.models.resnet50(), torch.randn(1, 3, 3, 1000).cuda())
+            ]
+            # Enable determinism in cudnn operators
+            for (model, inp) in models_to_test:
+                with torch.backends.cudnn.flags(
```

Comment:
Is there a global flag that we can set for all the tests in the same file? If we have more tests, we probably want to have the same setup.

Reply:
Unfortunately simply doing the following:
```
torch.backends.cudnn.benchmark = False
torch.backends.cudnn.deterministic = True
```

ran into some assertion errors, so we weren't able to just do that. This context manger has the benefit of setting/unsetting those variables as well so they're not unintentionally set.